TensorLib/Tensor.lean

```diff
@@ -0,0 +1,135 @@
+import Mathlib.Tactic
+namespace TensorLib
+section TensorData
+
+variable (a: Type)
```

Comment:
I had it like this originally, but we don't use it anywhere. Do you think there's value to it? For example, if we get the `TensorElement` restrictions on the type for free then it may be worth it.

Reply:
I was planning on adding the restriction anywhere I actually need it. Maybe that's essentially everwhere at some point so I should just restrict right off the bat.